core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java

```diff
@@ -957,6 +963,43 @@ public long getRetainedSizeInBytes()
         return Optional.of(Collections.unmodifiableList(result));
     }
 
+    /// Return smaller representation of this a discrete value set.
+    ValueSet compact()
+    {
+        if (!isDiscreteSet()) {
+            return this; // TODO: support also non-discrete sets
+        }
+        Type type = getType();
+        Ranges ranges = getRanges();
+        List<Range> result = new ArrayList<>();
+
+        Optional<Range> current = Optional.empty();
```

Comment:
If we initialised `current` to the first value in `ranges.getOrderedRanges()`, could we avoid using `Optional` here ?

Reply:
Sounds good, will fix.